buildenv/jenkins/JenkinsfileBase

```diff
@@ -538,40 +538,25 @@ def runTest( ) {
 			RUNTEST_CMD = "./openjdk-tests ${TARGET} ${CUSTOM_OPTION}"
 			for (int i = 0; i < ITERATIONS; i++) {
 				echo "ITERATION: ${iterationCount}/${ITERATIONS}"
-				if (env.BUILD_LIST == 'openjdk') {
-					if (env.SPEC.startsWith('linux_x86-64')) {
-						wrap([$class: 'Xvfb', autoDisplayName: true]) {
-							def DISPLAY = sh (
-								script: 'ps -f  | grep \'[X]vfb\' | awk \'{print \$9}\'',
-								returnStdout: true
-							).trim()
-							env.DISPLAY = "${DISPLAY}"
-							echo "env.DISPLAY is ${env.DISPLAY}"
-							makeTest("${RUNTEST_CMD}")
-						}
-					} else {
-						makeTest("${RUNTEST_CMD}")
-					}
-				} else if (env.BUILD_LIST.startsWith('jck')) {
-					if (env.SPEC.startsWith('linux')) {
-						wrap([$class: 'Xvfb', autoDisplayName: true]) {
-							def DISPLAY = sh (
-								script: 'ps -f  | grep \'[X]vfb\' | awk \'{print \$9}\'',
-								returnStdout: true
-							).trim()
-							env.DISPLAY = "${DISPLAY}"
-							echo "env.DISPLAY is ${env.DISPLAY}"
-							makeTest("${RUNTEST_CMD}")
-						}
-					} else if (env.SPEC.startsWith('aix')) {
-						sh "nohup /usr/bin/X11/X -force -vfb -x abx -x dbe -x GLX :0 &"
-						env.DISPLAY = ":0"
+				// Starting an X11 server
+				if ((env.BUILD_LIST == 'openjdk' || env.BUILD_LIST.startsWith('jck')) && env.SPEC.startsWith('aix')) {
```

Comment:
Actually wondering if we create a method called setEnvDisplay(env.BUILD_LIST, env.SPEC) as we will also want to set DISPLAY for osx.  We could potentially just set it no matter what BUILD_LIST is defined, as its presumably benign and does not interfere with tests that do not need it.  In that case, its essentially a check for SPEC and then running the appropriate commands to accomplish the setting of the DISPLAY variable on the machine.

```
setEnvDisplay (env.SPEC) {
    if (env.SPEC.startsWith('aix')) {
        sh "nohup /usr/bin/X11/X -force -vfb -x abx -x dbe -x GLX :0 &"
					env.DISPLAY = ":0"
    } else if (env.SPEC.startsWith('linux')) {
        wrap([$class: 'Xvfb', autoDisplayName: true]) {
						def DISPLAY = sh (
							script: 'ps -f  | grep \'[X]vfb\' | awk \'{print \$9}\'',
							returnStdout: true
						).trim()
						env.DISPLAY = "${DISPLAY}"
    } else if (env.SPEC.startsWith('osx') {
        // what ever commands needed there
    }
    echo "env.DISPLAY is ${env.DISPLAY}"
}
```
then call it at line 542 and follow with the call to makeTest

I believe the only reason that openjdk jdk_awt tests are not running on other platforms is this DISPLAY issue, so it will be good for us to run them against your branch changes to see if they look healthier.  (in other words, the check for && env.SPEC.startsWith('linux_x86-64') would no longer be needed, and we can re-enable targets on other platforms with DISPLAY set for all platforms).

Reply:
Okay so Ive removed the BUILD_LIST check and reduced the condition for SPEC to either aix or linux. It looks alot tidier and more manageable.

I havent added the commands needed to run an x11 server for osx as I do not have them. Is there already an open issue regarding running x11 tests on osx?

Reply:
For reference for what commands used by openjdk to set DISPLAY, see here: https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/test/start-Xvfb.sh (or [jdk11u version of start-Xvfb.sh](https://github.com/AdoptOpenJDK/openjdk-jdk11u/blob/master/test/jdk/start-Xvfb.sh)) though don't know the details of how they expect test machines to be configured.